config/routes.rb

```diff
@@ -1,11 +1,13 @@
 Rails.application.routes.draw do
 
+
   devise_for :users
   resources :users, only: [:update]
    
    resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```

Comment:
When you say nest under post, do you mean like this: resources :post, comments, [:index]
and topics would look like this: resources :topics, posts do